### DISCUSSION TITLE
Lilygo T-relay S3 asennus

### DISCUSSION THREAD

**Discussion by Peltoni** [Netgalleria/arska-node#44] [Q&A] [answered]
Lähdin noviisina kokeilemaan asennusta ja näinhän siinä kävi ettei ohjeista huolimatta onnistunut. Yritin siis web-installerilla asentaa ESP32-S3 softaa. USB-TTL muunnin asentui ok mutta sen jälkeen tökkäsi. Kokeilin asettaa laitteen manuaalisesti ”ohjelmointi”-tilaan, resetoida, bootata jne ja kaikkea näitä eri järjestyksessä. Tulos aina sama: web-installer antaa virheilmoituksen mikä sanoo että Installation failed. Mikä avuksi?
![IMG_0500](https://github.com/user-attachments/assets/9d0ea243-62fa-47c7-b0ce-581e30b75b0c)

**Comment by Olli69** [comment, marked as answer]
Arskan webbiasennuksessa käytetään ESPHome:n asennuskomponenttejä, joiden virheilmoituksen olet saanut. Voisit kokeilla sivulla https://esphome.github.io/esp-web-tools/ onnistuuko ESPHomen asennus laitteella. Jos ei onnistu, niin voisi johtua esim. kaapelista, tietokoneen USB-ajureista, USB-TTL-ohjaimesta  (tsekkaa myös että TX tai RX menee ristiin, jos olet joutunut vetämään piuhat yksitellen), kortin tilasta (eli pitäisi saada jollain ilveellä ohjelmontitilaan). Jos saat ESPHomen asentumaan, niin silloin pitäisi mennä  Arskan asennussivulta Arskakin.

**Comment by mmk9876** [comment]
Hei,
Hommasin Lilygo T-relay S3 kortin, jota en saa nyt toimimaan. Sain asennussivun ohjeilla käsittääkseni asennuksen onnistumaan ja yhdistettyä wifiin. Kun laitan selaimeen IP-osoitteen, pääsen sisään ja saan muutettua asetuksia, laitoin Eleringin kautta Suomen hintatiedon haun. Dashboardille tulee ilmoitus, että Cannot get price data from Elering. Kun yritän manuaalisesti kytkeä releitä, dashboardilla status muuttuu kyllä, mutta relekortilla ei tapahdu mitään. Mitähän tälle olisi tehtävissä?

Hyvä, että tällaista kaikille avointa kehitystä tehdään, kiitos siitä.

**Reply by Olli69** [reply]
Moi,
Tämän päivän hintatietoja ei jostain syystä ole saatavissa Eleringiltä. Entso-E:stä tulee, mutta sitä varten tarvitaan API-avain. Yleensä Eleringin kanssa on ollut tosi vähän ongelmia. Avasin uuden issuen [https://github.com/Netgalleria/arska-node/issues/45](https://github.com/Netgalleria/arska-node/issues/45) johon voi raportoida mahdollisia hintaongelmia ja seurata niiden tilaa

**Reply by mmk9876** [reply]
Kiitos vastauksesta. Sain APIn Entso-E:stä, nyt toimii hintatiedon haku. Releiden ohjaukset eivät edelleenkään toimi manuaaliohjauksella, vaikka status dashboardissa muuttuu kyllä tilaan _Up (manual schedule)_, mistä tiedän onko vika uudessa relekortissa vai voiko joku asetus olla vielä pielessä?

**Reply by Olli69** [reply]
Oletko asettanut Advanced Settingseissä kyseisen kortin profiilin?
<img width="561" alt="image" src="https://github.com/user-attachments/assets/362525ad-5dc2-4aec-819e-20023e8989b1">

**Reply by mmk9876** [reply]
Olen asettanut tuon profiilin. 
Pitäisikö tässä kortissa palaa jotain merkkivaloja kun virta on syötettynä? Ihan ensimmäistä kertaa kun laitoin usb:n kiinni, paloi/välkkyi vihreä ja punainen led (CH7 ja CH8) hetken aikaa, mutta Arskan asennuksen jälkeen ne ei ole enää palaneet. Tarkistin vielä yleismittarilla kaikki 3v ja 5v syötöt ja maadoitukset, ne ovat ainakin ok.

**Reply by Apulanta** [reply]
Täällä ei päivitys nuo Entso-E hinta tiedot ... API avain kyllä on..

**Reply by mmk9876** [reply]
Onhan sulla valittuna valikosta Finland, Fingrid nimen omaan Entso-E:n alta eikä Eleringin?
![Näyttökuva 2024-12-05 124625](https://github.com/user-attachments/assets/59b5ef84-852f-4b59-825c-84663a3ea404)

**Reply by Apulanta** [reply]
Juu on tuo " Finland, Fingrid " valittuna ... tuuli ennuste tulee ja PV aurinko.. kyllä.. tuntihinnat ei tule...

**Reply by Apulanta** [reply]
> Juu on tuo " Finland, Fingrid " valittuna ... tuuli ennuste tulee ja PV aurinko.. kyllä.. tuntihinnat ei tule...

Nyt hinnat päivittyi kun tein uudelleen käynnistyksen ja välimuistin tyhjennys..

**Reply by Olli69** [reply]
@mmk9876, jos kortissa ei ole [HAN P1 -lukijaa](https://robomaa.fi/index.php?route=product/product&product_id=8874),  niin valoja siinä ei juuri taida olla. Mutta jos laitat kanavan päälle manuaalisesti niin silloin kyllä relen sininen ledi pitäisi palaa.

Jäin miettimään, että en ole juurikaan testannut Arskaa  ilman HAN P1 -lukijaa, mutta ei kyllä tule mieleen miksei toimisi.

**Reply by mmk9876** [reply]
Ei ole HAN P1 -lukijaa. Nuo siniset ledit ei syty kun ei releetkään kytkeydy.

**Reply by Olli69** [reply]
@mmk9876 , tämä voi johtua siitä, että sinulla uusi revisio 1.1 laitteesta. Löysin [infoa](https://github.com/Xinyuan-LilyGO/LilyGo-T-Relay/issues/21), jonka mukaan tässä releiden ohjauksessa on muutos verrattuna ensimmäiseen rautaversioon eli pitäisi ajaa gpio04 alas bootissa. Voitko lähettää mulle sähköpostia olli@netgalleria.fi, niin lähetän sinulle koeversion. Itsellä ei ole tuota rev 1.1 rautaa tähän hätään käytössä.

**Reply by Olli69** [reply]
Eli päivityksen jälkeen saatiin myös @mmk9876 :n Arska toimimaan. LilyGo 6-porttisen ESP32S3 rev 1.1 koskeva päivitys 1.1.0-beta12 on nyt julkaistu myös [Arskan asennussivulla](https://iot.netgalleria.fi/arska-install/).

**Comment by Corn-101** [comment]
@Peltoni miten sait asennuksen toimimaan? Minulla tulee sama virheilmoitus.

**Reply by Peltoni** [reply]
En saanut, heitin laitteet mäelle ja ostin shellyjä tilalle 😀

**Reply by Corn-101** [reply]
Löysin vian. Asennus ei tähän S3:n onnistu Windows 7 koneelta. Vanhempaan Lilygo:n onnistuu seiskalla, joten aloin jo epäilemään omaa mielenterveyttä. Oliko sullakin mahdollisesti seiska, millä yritit päivitellä?

**Reply by Peltoni** [reply]
Taisipa hyvinkin olla. Kaikkea muuta tulikin koitettua ja melko moneen kertaan… mutta hyvä että keksit ja homma skulaa! Itse tyytyväinen suunnanmuutokseen, shellyt sopii paremmin omille hermoille ja taitotasolle.

**Reply by Olli69** [reply]
@Corn-101, Hyvä jos vian syy selvisi. Lisäsin asennussivun lisätietoihin, että Windows 7/ESP32S3 -kombon kanssa ongelmia. Löysitkö muuten netistä tästä jonkun maininnan vaiihan kokeilemalla?

**Reply by Corn-101** [reply]
@Peltoni Itse olen käyttänyt Arskaa jo yli vuoden ja olen kyllä pirun tyytyväinen. Olli korjailee bugit ja tekee järkevät kehitysehdotukset vieläpä todella nopeasti ja softa on luotettava ja helppokäyttöinen. Enkä enää keksi edes monimutkaisempia sääntöjä ohjauksiin, mitä tällä ei voisi jo suorittaa. 

@Olli69 En löytänyt mitään tähän viittaavaan netistä. Viimeiseksi oljenkorreksi, kun tuntikaupalla yritin tuolla seiskakoneella päivittää S3:sta, jolla juuri olin onnistunut useampaan kertaan vanhempaa TTGO:ta päivittämään, niin päätin vielä kokeilla toisella läppärillä, missä uudempi winkkari. Ja sehän meni heittämällä läpi. Hyvä, kun lisäsit tuon maininnan tuosta kombosta, ettei muiden tarvii samaan petäjään päätänsä hakata. :)
